extensions/sql-migration/src/scripts/SHIR-auto-configuration.ps1

```diff
@@ -0,0 +1,609 @@
+	}
+}
+
+################################### Download, install and configure IR ##################################
+Function Install-IR {
+	Write-OutputAndLog "Checking if Integration Runtime is already installed on this machine..."
+	$installedIRVersion = Get-InstalledShirVersion
+
+	if ($null -ne $installedIRVersion) {
+		Write-OutputAndLog "Found Integration Runtime installed on the machine."
+
+		# Perform validation checks for installed IR
+		if ($installedIRVersion -ne $Global:LatestIRVersion) {
+			Write-ErrorAndLog "Installed Integration Runtime does not meet the checks for configuring. Please read the documentation for more information."
+			return
+		}
+		Write-OutputAndLog "Installed Integration Runtime meets the requirements for configuring."
+
+		if (-not (Test-InternetConnectivity)) {
+			return
+		}
+
+		# Configure SHIR - Register IR with the auth key
+		Register-IntegrationRuntime -AuthKey1 $AuthKey1 -AuthKey2 $AuthKey2
+	}
+	elseif (Test-CanInstallIR) {
+		$downloadFolder = Join-Path (New-Object -ComObject Shell.Application).NameSpace('shell:Downloads').Self.Path "SHIR-Installer"
+		$downloadUrl = "aka.ms/downloadLatestIR"
+
+		# Download the installer
+		$packageDownloadPath = Get-Package -packageName "IntegrationRuntime_$Global:LatestIRVersion.msi" `
+			-url $downloadUrl `
+			-downloadFolder $downloadFolder `
+			-Force
+
+		# Installs the IR msi
+		$installationWasSuccessful = Install-MsiPackage -packageName "IntegrationRuntime" `
+			-packagePath $packageDownloadPath `
+			-installationVerificationCallback { return $null -ne (Get-InstalledShirVersion) }
+
+		if ($installationWasSuccessful) {
+			# Configure SHIR - Register IR with the auth key
+			Register-IntegrationRuntime -AuthKey1 $AuthKey1 -AuthKey2 $AuthKey2
+		}
+	}
+}
+
+Function Get-LogFilePath {
+	<#
+    .DESCRIPTION
+    Creates the log file path inside the user profile and returns the path.
+
+    .OUTPUTS
+    Log file path
+    #>
+
+	# Get the current user's home directory using the environment variable USERPROFILE
+	$home_dir = $env:USERPROFILE
+
+	# Create the .shir folder inside the home directory using the Join-Path cmdlet
+	$shir_dir = Join-Path $home_dir "shir"
+	New-Item -ItemType Directory -Path $shir_dir -Force
+
+	# Create the logs.txt file inside the .shir folder using the New-Item cmdlet
+	$logs_file = Join-Path $shir_dir "logs.txt"
+	New-Item -ItemType File -Path $logs_file -Force
+
+	Write-OutputAndLog ("Log file generated: " + $logs_file)
+
+	return $logs_file
+}
+
+################################### Pre-installation checks ##################################
+Function Test-CanInstallIR {
+	<#
+    .DESCRIPTION
+    Performs pre-installation checks to ensure that SHIR can be installed, if any check fails a terminating error is thrown.
+
+    .OUTPUTS
+    [bool]: True if pre-installation validation is passed else false.
+    #>
+	try {
+		if (-not [Environment]::Is64BitOperatingSystem) {
+			throw "Pre-requisite not met - OS is not 64 bits."
```

Comment:
Consider changing this to "64-bit OS is a pre-requisite" (Or something similar - consult PMs)

Reply:
Prerequisites are not met. 64-bit operating system is required.

Reply:
Updated!